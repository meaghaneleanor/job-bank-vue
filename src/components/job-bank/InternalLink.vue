<template>
  <div>
    <router-link :to="translateRoute(url)">
     {{ linkText }}
    </router-link>
  </div>
</template>

<script>
import { translate } from '../../mixins/translate.js';

export default {
  props: {
    url: {
      type: String,
      required: true
    },
    linkText: {
      type: String,
      required: true
    },
  },
  mixins: [translate]
};
</script>
