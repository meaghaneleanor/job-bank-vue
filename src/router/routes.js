import routeTranslations from '../translations/route-translations.js';

import Main from '../components/job-bank/Main.vue';
import Page404 from '../components/common/Page404.vue';

/*
  Because the route paths are defined in English, the search
  for the route translations will always be English, to retrieve the
  alias path in french
*/
const translatedAlias = function(path) {
  return routeTranslations['en'][path];
};

export default [
  {
    name: 'job-bank-search',
    path: '/page/ontario-job-bank',
    alias: translatedAlias('/page/ontario-job-bank'),
    component: Main
  },
  {
    name: 'job-bank-search-results',
    path: '/page/ontario-job-bank/search-results',
    alias: translatedAlias('/page/ontario-job-bank/search-results'),
    component: Main,
  },
  {
    name: 'page-not-found',
    path: '/*',
    component: Page404
  }
];
