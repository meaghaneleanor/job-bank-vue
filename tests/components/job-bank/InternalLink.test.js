import {shallowMount, createLocalVue} from '@vue/test-utils';
import {$t, i18nMock} from '../../test.utils.js';
import VueRouter from 'vue-router';

import routes from '../../../src/router/routes';

import InternalLink from '../../../src/components/job-bank/InternalLink.vue';

const localVue = createLocalVue();
const router = new VueRouter({
  mode: 'history',
  routes,
  options: { routes }
});
localVue.use(router);

const defaultOptions = {
  propsData: {
    url: '/page/jobs',
    linkText: 'Job Link',
  },
  mocks: {
    $t,
    i18nMock
  },
  localVue,
  router,
  stubs: ['router-link']
};

const createComponent = (options=defaultOptions) => shallowMount(InternalLink, options);

describe('InternalLink.vue', () => {
  describe('snapshot', () => {
    it('should display the expected html', () => {
      let wrapper = createComponent();
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
