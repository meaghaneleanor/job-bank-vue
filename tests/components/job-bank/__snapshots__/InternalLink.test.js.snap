// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InternalLink.vue snapshot should display the expected html 1`] = `
"<div><router-link-stub to=\\"/page/ontario-job-bank\\" tag=\\"a\\" event=\\"click\\">
    Job Link
  </router-link-stub></div>"
`;
